modules/series-service-impl/src/main/java/org/opencastproject/series/impl/persistence/SeriesServiceDatabaseImpl.java

```diff
@@ -303,13 +316,20 @@ public DublinCoreCatalog storeSeries(DublinCoreCatalog dc)
     DublinCoreCatalog newSeries = null;
     try {
       tx.begin();
-      SeriesEntity entity = getSeriesEntity(seriesId, em);
-      if (entity == null) {
+      SeriesEntity entity = getPotentiallyDeletedSeriesEntity(seriesId, em);
+      if (entity == null || entity.isDeleted()) {
+        // If the series existed but is marked deleted, we completely delete it
+        // here to make sure no remains of the old series linger.
+        if (entity != null) {
+          this.deleteSeries(seriesId);
```

Comment:
Just ran into this while testing my migration scripts for Bern - this doesn't actually seems to work, since deleteSeries() would only mark the entity as deleted again, not remove it (found no method to do this), but mostly because this throws a NotFound for entities which are already marked as deleted. To test this, just create a series, delete it, and then try to recreate a series with the same id. Since you know how this feature is supposed to work, can you take a look at this?